src/worker/helpers.go

```diff
@@ -166,46 +166,192 @@ func initOLDir(olPath string, dockerBaseImage string, newBase bool) (err error)
+		if err := os.WriteFile(kill, []byte(fmt.Sprintf("%d", 1)), os.ModeAppend); err != nil {
+			// Log an error if killing processes in the cgroup fails.
+			fmt.Printf("Could not kill processes in cgroup: %s\n", err.Error())
+		}
+		for _, file := range files {
+			if strings.HasPrefix(file.Name(), "cg-") {
+				cg := filepath.Join(cgRoot, file.Name())
+				fmt.Printf("Attempting to remove %s\n", cg)
+				if err := syscall.Rmdir(cg); err != nil {
+					// Return an error if removing a cgroup fails.
+					return fmt.Errorf("could not remove cgroup: %s", err.Error())
+				}
+			}
+		}
+		if err := syscall.Rmdir(cgRoot); err != nil {
+			// Log an error if removing the cgroup root directory fails.
+			fmt.Printf("Could not remove cgroup root: %s\n", err.Error())
+		}
+	}
+
+	// Clean up mounts associated with sandboxes
+	dirName := filepath.Join(olPath, "worker", "root-sandboxes")
+	fmt.Printf("Attempting to clean up mounts at %s\n", dirName)
+
+	if files, err := os.ReadDir(dirName); err != nil {
+		// Return an error if the mount root directory cannot be found.
+		return fmt.Errorf("could not find mount root: %s", err.Error())
+	} else {
+		for _, file := range files {
+			path := filepath.Join(dirName, file.Name())
+			fmt.Printf("Attempting to unmount %s\n", path)
+			if err := syscall.Unmount(path, syscall.MNT_DETACH); err != nil {
+				// Return an error if unmounting fails.
+				return fmt.Errorf("could not unmount: %s", err.Error())
+			}
+			if err := syscall.Rmdir(path); err != nil {
+				// Return an error if removing the mount directory fails.
+				return fmt.Errorf("could not remove mount dir: %s", err.Error())
+			}
+		}
+	}
+
+	// Attempt to unmount the main mount directory
+	if err := syscall.Unmount(dirName, syscall.MNT_DETACH); err != nil {
+		// Log an error if unmounting the main directory fails.
+		fmt.Printf("Could not unmount %s: %s\n", dirName, err.Error())
+	}
+
+	// Remove the worker.pid file
+	if err := os.Remove(filepath.Join(olPath, "worker", "worker.pid")); err != nil {
+		// Return an error if removing worker.pid fails.
+		return fmt.Errorf("could not remove worker.pid: %s", err.Error())
+	}
+
+	return nil
+}
+
+// generalCleanup checks the state of Open Lambda and performs appropriate cleanup actions.
+// It calls gracefulCleanup for running instances and dirtyCleanup for instances in DirtyShutdown state.
+func generalCleanup(olPath string) error {
+	state, pid, err := checkState(olPath)
+	if err != nil {
+		return fmt.Errorf("failed to check OL state: %s", err)
+	}
+
+	// Retrieve the process handle for the identified PID.
+	p, _ := os.FindProcess(pid)
+
+	switch state {
+	case Running:
+		err := gracefulCleanup(p)
+		if err != nil {
+			return fmt.Errorf("failed to gracefully cleanup OL: %s", err)
+		}
+	case DirtyShutdown:
+		dirtyCleanup(olPath)
+	case Unknown:
+		return fmt.Errorf("unknown state detected")
+	}
+
+	return nil
+}
+
+// stopOL attempts to gracefully stop Open Lambda (OL) and handles errors.
```

Comment:
These functions seem like pretty simple wrappers.  Can we get rid of them and call generalCleanup or gracefulCleanup directly?

Reply:
stopOL is used in multiple different places. I think the name of generalCleanup might be a bit misleading because it also stops the process if it is running.